pkg/utils/sets/sets.go

```diff
@@ -0,0 +1,105 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sets
+
+import (
+	"fmt"
+	"math"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// Set is a logical set of string values for the requirements.
+// It supports representations using complement operator.
+// e.g., if C={"A", "B"}, setting complement = true means
+// C' contains every possible string values other than "A" and "B"
+type Set struct {
+	values     sets.String
+	Complement bool
```

Comment:
does this need to be exported? Would you ever create a set and then configure it to be a complement set?

Reply:
Good catch! Done!